api/src/main/java/org/eclipse/microprofile/concurrent/ThreadContext.java

```diff
@@ -240,6 +240,44 @@ public static Builder builder() {
      * scopes. An empty/default CDI context means that the thread does not have
      * access to the scope of the session, request, and so forth that created the
      * contextualized action.
+     * 
+     * For example, consider the following <code>&#64;RequestScoped<code> resource:
+     * 
+     * <code><pre>
+    &#64;RequestScoped
+    public class MyRequestScopedBean {
+        public String getState() {
+          // returns some request-specific information to the caller
+        }
+    }
+    </pre></code>
+     * 
+     * When CDI context is propagated, the request's state will also
+     * be available to the contextualized work: <code><pre>
+    ManagedExecutor exec = ManagedExecutor.builder()
+        .propagated(ThreadContext.CDI, ThreadContext.APPLICATION)
+        .build();
+    
+    &#64;Inject
+    MyRequestScopedBean requestBean;
+    
+    &#64;GET
+    public void foo() {
+        exec.supplyAsync(() -> {
+            String state = reqBean.getState();
+            // do some work with the request state
+        }).thenApply(more -> {
+            // request state also available in future stages
+        });
+    }
+    </pre></code>
+     *
+     * If CDI context is 'cleared', the state of any Request, Session, or Conversation
```

Comment:
I'd drop the notion of restoring contexts, that's implied by the nature of propagation (as is cleanup after you propagate it). Furthermore, we need to stress out that `cleared` means you will have the same contexts active but with no state it them.

_If CDI context is `cleared`, currently active contexts will still be available to the contextualized task, but their state will be erased._